### PR TITLE
[F] Added the UniqueByArgument constraint to the async ExportJob (ENT-1265)

### PR DESCRIPTION
**Pull request by Ceiu**
- Added the UniqueByArgument constraint to ExportJob, preventing it
  from being queued if another is already active for the same
  consumer
- Changed a few job arguments to metadata or explicit job
  configuration as appropriate

### CODE REVIEW

**Comment by Ceiu** [comment]
Note that this PR doesn't completely finish the ExportJob task. We still need to wire it up, clean up the way the job is scheduled, and possibly fix the way the job key is registered. These will come after the AsyncJobStatus DTO is added.

File: server/src/main/java/org/candlepin/async/tasks/ExportJob.java [hunk 1]
```diff
@@ -77,7 +80,7 @@ public ExportJob(ManifestManager manifestManager) {
     @Override
     public Object execute(final JobExecutionContext jdata) throws JobExecutionException {
         final JobDataMap map = jdata.getJobData();
-        final String consumerUuid = map.getAsString(JobStatus.TARGET_ID);
+        final String consumerUuid = map.getAsString(CONSUMER_KEY);
```
Comment (mstead) [inline server/src/main/java/org/candlepin/async/tasks/ExportJob.java]: :+1:  Way more explicit.

File: server/src/main/java/org/candlepin/async/tasks/ExportJob.java [hunk 2]
```diff
@@ -113,16 +116,14 @@ public static JobBuilder scheduleExport(
         final Map<String, String> extensionData
     ) {
         return JobBuilder.forJob(JOB_KEY)
```
Comment (mstead) [inline server/src/main/java/org/candlepin/async/tasks/ExportJob.java]: **A NOTE, BUT NOT FOR THIS PR:**
Not overly worried if this gets changed, but, the method scheduleExport is a little misleading as it isn't really scheduling anything; it is just creating a builder. A better name... not sure. :)

**Comment by mstead** [comment]
Looks good, waiting on tests.

File: server/src/main/java/org/candlepin/async/tasks/ExportJob.java (hunk 2 above)
Comment (Ceiu) [reply server/src/main/java/org/candlepin/async/tasks/ExportJob.java]: This is definitely on my radar for the second part of this. I was never terribly thrilled with scheduling a job by returning an object from a resource and letting a filter pick it up later.

Need to come up with something that's easy to use and a little more clear in the intent.

**Pull request merged**